### ISSUE TITLE
Rule avoid_type_to_string not working

### ISSUE THREAD

**Issue by tmechelke** [dart-lang/linter#2597]
The rule avoid_type_to_string is not working for me.

Here are the steps to reproduce:

1) `dart --version`
Dart SDK version: 2.12.3 (stable) (Wed Apr 14 11:02:39 2021 +0200) on "windows_x64"

2) `dart create --template package-simple example_lib`

3) Replace content of analysis_options.yaml with the following:
```
linter:
  rules:
    avoid_type_to_string: true
```

4) Append the following to lib/example_lib.dart:
```
class A {}

void f() {
  A().runtimeType.toString();
}
```

5) `dart analyze`
Expected result: a message regarding avoid_type_to_string
Actual result: no issues found

**Comment by goderbauer** [comment]
I was just running into this as well, the BAD example given in the docs [1] also doesn't produce any lints:

```dart
void bar(Object other) {
  if (other.runtimeType.toString() == 'Bar') {   // LINT expected
    doThing();
  }
}

Object baz(Thing myThing) {
  return getThingFromDatabase(key: myThing.runtimeType.toString());   // LINT expected
}

// ---

void doThing() {

}

Object getThingFromDatabase({String? key}) {
  return 'Foo';
}

class Thing {

}
```

[1] https://dart-lang.github.io/linter/lints/avoid_type_to_string.html

**Comment by pq** [comment]
@a14n: any thoughts?

**Comment by a14n** [comment]
@pq I'm not the author of this lint ;)

**Comment by pq** [comment]
Sorry!  I was confused w/ `no_runtimeType_toString`!  😬  Apologies!

Looks like that one was a contribution from @nscobie (https://github.com/dart-lang/linter/pull/2209).  They're on to other stuff though so we'll likely have to look at it.

**Comment by nscobie** [comment]
Hi all, sorry to hear this lint is having issues.

I can definitely repro this _not_ working with `dart analyze` using Dart SDK version `2.12.3`, as reported.

However, it _does_ appear to still be working on my end when running `dart test -N avoid_type_to_string` in this repository.

1. When testing in this repository, it successfully reports all violations for the current [test cases](https://github.com/dart-lang/linter/blob/master/test_data/rules/avoid_type_to_string.dart), the BAD example listed in the docs, and the repro case reported in this issue.
2. This holds true for both the current HEAD (commit [de386a3](https://github.com/dart-lang/linter/commit/de386a3cbb01a13b58cab4f777ed1e76dc3cb0e4)), and linter version [0.1.129](https://github.com/dart-lang/linter/releases/tag/0.1.129) (the one [used in Dart SDK 2.12.3](https://github.com/dart-lang/sdk/blob/master/CHANGELOG.md#2123---2021-04-14)).

Could this be an issue with rolling into/configuring in the SDK/analyzer? I am not at all familiar with that process, or the overall interplay between the SDK and linter packages.

Unfortunately I won't have much availability to help with this until my exams are over in a few weeks.

**Comment by oprypin** [comment]
FWIW this issue still applies - this lint never finds any violations, as far as I see.
